### ISSUE TITLE
Escape inside regex group results in spurious SC1009/SC1072/SC1073

### ISSUE THREAD

**Issue by CyberShadow** [koalaman/shellcheck#1077]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1009, SC1073, SC1072
- My shellcheck version (`shellcheck --version` or "online"): online and 0.4.7
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit
- [ ] It's not reproducible on shellcheck.net, but I think that's because it's an OS, configuration or encoding issue

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
[[ $1 =~ ^(a\ b)$ ]]
```

#### Here's what shellcheck currently says:

```
In test.sh line 2:
[[ $1 =~ ^(a\ b)$ ]]
         ^-- SC1009: The mentioned parser error was in this regex.
          ^-- SC1073: Couldn't parse this regex grouping.
             ^-- SC1072: Unexpected . Fix any mentioned problems and try again.
```

#### Here's what I wanted or expected to see:

Nothing.

This appears to be a regression - previous ShellCheck versions did not complain about similar code.

**Comment by koalaman** [comment]
Nice catch. This should now be fixed.

**Comment by CyberShadow** [comment]
Thanks!
